### PR TITLE
Add XML Docs for WithArgs()

### LINKED ISSUE #1294: WithArgs method is missing XML doc comments
image

### PR DESCRIPTION
**Pull request by ericmutta**
The wording in the XML docs is based on my cursory review of [issue 177](https://github.com/dotnet/aspire/issues/177) and [PR 1121](https://github.com/dotnet/aspire/pull/1121).

I should add that I only started reviewing Aspire a few hours ago, so apologies if the XML docs aren't accurate - I just figured the best way to get started contributing is to jump right in (CC: @danmoseley, @DamianEdwards).

#Fixes https://github.com/dotnet/aspire/issues/1294

### CODE REVIEW

File: src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs [hunk 1]
```diff
@@ -81,6 +81,13 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Adds the arguments to be passed to a container resource when the container is started.
```
Comment (DamianEdwards) [inline src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: This extension method isn't currently constrained to `ContainerResource` (see line 91 below) so I'm not sure what it does when used with other resource types. We'll likely need to check that and update the doc comments appropriately.
@eerhardt @davidfowl

Comment (mitchdenny) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: I had a look through the implementation. It is definitely container specific.

Comment (davidfowl) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: It's not specific right now. It also works for executables.

Comment (DamianEdwards) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: @mitchdenny @davidfowl so, which is it? Constraining this method to `where T : ContainerResource` seems wrong if this indeed works for executables too.

Comment (davidfowl) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: It's both executables and containers.

Comment (mitchdenny) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: David is correct. I must have been hallucinating.

Comment (mitchdenny) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: Actually I don't think that `WithArgs` is fully baked. I am wondering if it was actually intended to be private. @BrennanConroy

Comment (davidfowl) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: https://github.com/dotnet/aspire/blob/91034428ea2f6058ed572920059b680e7d7cdcc0/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L572-L579

Comment (mitchdenny) [reply src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs:85]: OK find references in VS might be having an issue ;)

**Comment by davidfowl** [comment]
@mitchdenny can you take this and backport it once it's in?

**Review by mitchdenny — APPROVED** [review]

**Pull request merged**